probe_src/probe_py/test_probe.py

```diff
@@ -0,0 +1,253 @@
+from typer.testing import CliRunner
+import tarfile
+from .cli import app
+from . import parse_probe_log
+from . import analysis
+import pathlib
+import typing
+import networkx as nx
+import subprocess
+
+runner = CliRunner()
+
+def test_diff_cmd():
+    result = runner.invoke(app,["record","--make","diff","../flake.nix","../flake.lock"])
+    input: pathlib.Path = pathlib.Path("probe_log")
```

Comment:
```suggestion
    input = pathlib.Path("probe_log")
```

We don't need a type annotation here, because the type of input can be inferred from the signature of `pathlib.Path(...)`. When we have Mypy (future task on the list) that will become more clear.